src/shared/sam/cli/samCliLocalInvoke.ts

```diff
@@ -99,8 +99,7 @@ export class DefaultSamLocalInvokeCommand implements SamLocalInvokeCommand {
                 }
             })
 
-            if (!params.isDebug) {
-                this.channelLogger.logger.verbose('Local SAM App does not expect a debugger to attach.')
+            if (!params.isDebug || this.debuggerAttachCues.length === 0) {
```

Comment:
can't this be `!checkForDebuggerAttachCue`

Reply:
Not really, there's a chance that checkForDebuggerAttachCue has been altered by the time that line is reached

Reply:
Why not make it a function then?